clients/js/test/extensions/scaledUiAmountMint/amountToUiAmount.test.ts

```diff
@@ -0,0 +1,424 @@
+    decimals: decimals,
+    isInitialized: true,
+    freezeAuthority: defaultAddress,
+    extensions: hasScaledUiAmountConfig
+      ? [
+          {
+            __kind: 'ScaledUiAmountConfig',
+            authority: defaultAddress,
+            multiplier: config.multiplier || 2,
+            newMultiplierEffectiveTimestamp:
+              config.newMultiplierEffectiveTimestamp || ONE_YEAR_IN_SECONDS * 3,
+            newMultiplier: config.newMultiplier || 2,
+          },
+        ]
+      : [],
+  });
+  return populateMockAccount(data);
+}
+
+function createMockClockAccountInfo(unixTimestamp: number) {
+  const clockEncoder = getSysvarClockEncoder();
+  const data = clockEncoder.encode({
+    epoch: 0n,
+    epochStartTimestamp: 0n as UnixTimestamp,
+    leaderScheduleEpoch: 0n,
+    slot: 0n,
+    unixTimestamp: BigInt(unixTimestamp) as UnixTimestamp,
+  });
+  return populateMockAccount(data);
+}
+
+// GROUP 1: Basic functionality - standard mint without scaled UI extension
+test('should return the correct UiAmount when scaled ui amount config is not present', async (t) => {
+  const testCases = [
+    { decimals: 0, amount: BigInt(100), expected: '100' },
+    { decimals: 2, amount: BigInt(100), expected: '1' },
+    { decimals: 9, amount: BigInt(1000000000), expected: '1' },
+    { decimals: 10, amount: BigInt(1), expected: '1e-10' },
+    { decimals: 10, amount: BigInt(1000000000), expected: '0.1' },
+  ];
+
+  for (const { decimals, amount, expected } of testCases) {
+    const rpc = getMockRpc({
+      [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+      [mint]: createMockMintAccountInfo(decimals, false),
+    });
+    const result = await amountToUiAmountForMintWithoutSimulation(
+      rpc,
+      mint,
+      amount
+    );
+    t.is(result, expected);
+  }
+});
+
+// GROUP 2: Basic integer multiplier tests
+test('should return the correct UiAmount with scale factor of 2', async (t) => {
+  const testCases = [
+    { decimals: 0, amount: BigInt(100), expected: '200' },
+    { decimals: 1, amount: BigInt(100), expected: '20' },
+    { decimals: 10, amount: BigInt(10000000000), expected: '2' },
+  ];
+
+  for (const { decimals, amount, expected } of testCases) {
+    const rpc = getMockRpc({
+      [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+      [mint]: createMockMintAccountInfo(decimals, true, { multiplier: 2 }),
+    });
+
+    const result = await amountToUiAmountForMintWithoutSimulation(
+      rpc,
+      mint,
+      amount
+    );
+    t.is(result, expected);
+  }
+});
+
+test('should return the correct UiAmount with scale factor of 3', async (t) => {
+  const rpc = getMockRpc({
+    [CLOCK]: createMockClockAccountInfo(ONE_YEAR_IN_SECONDS * 2),
+    [mint]: createMockMintAccountInfo(10, true, { multiplier: 3 }),
+  });
+
+  const result = await amountToUiAmountForMintWithoutSimulation(
+    rpc,
+    mint,
+    BigInt(10000000000)
+  );
+  t.is(result, '3');
+});
```

Comment:
nit: it's fine to keep this, but it seems a bit redundant given the previous test is almost exactly the same thing, but more complete